pymeasure/instruments/thorlabs/elliptec_utils/base.py

```diff
@@ -0,0 +1,209 @@
+        else:
+            self.ask('ho1')
+
+    def move(self, position: int, move_type: str = 'abs') -> None:
+        """ Move the MotorChannel to an absolute or relative position in steps
+
+        :param position: value to move in steps
+        :param move_type: either 'abs' for an absolute move or something else for relative move
+        """
+        if move_type == 'abs':
+            command = 'ma'
+        else:
+            command = 'mr'
+        command += self.int_to_hex(position)
+        self.ask(command)
+
+    @staticmethod
+    def int_to_hex(number: int):
+        """ Convert integer to uppercase hexadecimal character """
+        return int_to_hex(number)
+
+
+class ElliptecController(Instrument):
+    """ An Elliptec Controller adding motor channels at given addresses
+
+    :arg channels: dictionary mapping from an integer as a string ('0', '1', ...) to a MotorChannel or inherited
+    objects
+    """
+
+    def __init__(self, adapter, name="Elliptec Rotator", klass=MotorChannel, address: Union[int, List] = 0, **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            **kwargs)
+
+        if address is not None:
+            self.add_motor_channel(klass, address)
+
+    def add_motor_channel(self, klass=MotorChannel, address: Union[int, List] = 0):
+        """ Add one or several MotorChannel instance (or subclassed) with a given address to the Motor channels
+
+        :param klass: A MotorChannel object or any inherited Motor (LinearChannel, RotatorChannel...)
+        :param address: integer or list of integer as Motor address
+        """
+        if not isinstance(address, Sequence):
+            address = [address]
+
+        for ad in address:
+            self.add_child(klass, str(ad))
+            try:
+                self.load_motor_info()
+            except VisaIOError:
+                raise ExternalDeviceNotFound
+
+    def load_motor_info(self, address: int = 0):
+        """ Asks motor for info and load response into properties """
+        self.channels[str(address)].load_motor_info()
+
+    def close(self):
+        """ Close the adapter communication, here the COM port"""
+        try:
+            self.adapter.close()
+        except:
+            pass
+
+
+def scan_for_devices(adapter, start_address=0, stop_address=0):
+    """ Scan for valid addresses at the given ElliptecController
+
+    :param adapter: Visa Adapter
+    :param start_address: scan from an address integer
+    :param stop_address: scan up to an address integer
+    """
+    channels = []
+    motor = ElliptecController(adapter, address=None)
+    for address in range(start_address, stop_address+1):
+        try:
+            motor.add_motor_channel(address=address)
+            channels.append(address)
+        except ExternalDeviceNotFound:
+            pass
+    motor.adapter.close()
+    return channels
+
+
+if __name__ == '__main__':
+
+    channels = scan_for_devices('ASRL19::INSTR', 0, 3)
```

Comment:
Is it for testing purposes? Then you could create a "test_xy_with_device.py" file. We have a method to indicate the device address for tests with a device: https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#device-tests

I see limited usage in the pymeasure library for a fixed address.